### PR TITLE
Feat: Add class_proba option to semantic segmentation post-processing

### PR DESCRIPTION
**Pull request by demoncoder-crypto**
Addresses #37715, this is my first implementation, if my direction is correct please do let me know I will iteratively fix everything

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @qubvel @nielsrogge

**Comment by matteot11** [comment]
Hi everyone, thanks for the work on this PR — I'm also interested in obtaining pixel-level class probabilities during post-processing for semantic segmentation.

I noticed that the current implementation does not compute true pixel-level probability distributions. In particular, this line:
```segmentation = torch.einsum("bqc, bqhw -> bchw", masks_classes, masks_probs)```
produces a tensor of shape [b, c, h, w] whose values are not guaranteed to form a valid probability distribution over classes at each pixel. Specifically, the following is False in general:
 ```torch.isclose(segmentation.sum(1), torch.ones_like(segmentation)).all()```.
This happens because each pixel-class score is a weighted sum over all queries, combining the objectness scores at a pixel with the class probabilities per query — but there's no guarantee this will sum to 1 across classes.

In my use case, I need the output to be a valid per-pixel class probability distribution (i.e., summing to 1 over classes), so I'm currently normalizing the result like this (and getting quite acceptable results in pratice):
```
segmentation = torch.einsum("bqc, bqhw -> bchw", masks_classes, masks_probs)
segmentation = segmentation / segmentation.sum(1, keepdim=True)
```
An alternative might be to apply softmax() over the class dimension after the einsum, but this tends to overly smooth the predictions, reducing confidence in the most likely classes.

Just wanted to share this in case it's helpful, and to see if others have encountered similar needs or solutions.
